client/ayon_houdini/plugins/create/create_hda.py

```diff
@@ -201,34 +194,63 @@ def create_instance_node(
                 parent_node = pane.pwd()
 
             to_hda = parent_node.createNode(
-                "subnet", node_name="{}_subnet".format(node_name))
+                "subnet", node_name="{}_subnet".format(node_name)
+            )
+
         if not to_hda.type().definition():
-            # if node type has not its definition, it is not user
-            # created hda. We test if hda can be created from the node.
+            # If the node's type lacks a definition, it isn't an user-created HDA.
+            # We test whether an HDA can be generated from this node.
             if not to_hda.canCreateDigitalAsset():
-                raise CreatorError(
-                    "cannot create hda from node {}".format(to_hda))
+                raise CreatorError("cannot create hda from node {}".format(to_hda))
 
             # Pick a unique type name for HDA product per folder path per project.
-            type_name = (
-                "{project_name}{folder_path}_{node_name}".format(
-                    project_name=get_current_project_name(),
-                    folder_path=folder_path.replace("/","_"),
-                    node_name=node_name
-                )
+            type_name = "{project_name}{folder_path}_{node_name}".format(
+                project_name=get_current_project_name(),
+                folder_path=folder_path.replace("/", "_"),
+                node_name=node_name,
             )
 
+            source_parm_template_group = to_hda.parmTemplateGroup()
+
+            output_index_list = set()
+            for node in [
+                node
+                for node in to_hda.allSubChildren()
+                if node.type().name() == "output"
+            ]:
+                output_index = node.parm("outputidx").eval()
+                output_index_list.add(output_index)
+
+            subnet_inputs_with_connections = set()
+            for subnet_input in to_hda.indirectInputs():
+                if subnet_input.outputs():
+                    subnet_inputs_with_connections.add(subnet_input)
```

Comment:
Could you add a comment above this as to what it is trying to solve?

Why do we care about the children output nodes?
Also, doesn't `allSubChildren()` recurse into potential other subnetworks and their children, etc. which doesn't sound like "outputs" we care about?

Reply:
> Why do we care about the children output nodes? Also, doesn't `allSubChildren()` recurse into potential other subnetworks and their children, etc. which doesn't sound like "outputs" we care about?

the recursive thought is a good one i will need to check this case. 

and we care about the child output nodes because that defines how many outputs your subnet will have (i mean there index dose but you cant have 2 output nodes with the same index) i use this to determine how many ouputs your hda should have.

Reply:
> Shouldn't we also consider existing outputs (e.g. connections to outside of my current selection) even if I don't have a `output` node for them already?
> 
> It may be worth attaching example hip files you're testing with.

if i understood the code right this should just trigger for subnet types not for node selections ? 
and in an subnet the inputs are determined by the input nodes that have a connection and the outputs by the output nodes

Reply:
> if i understood the code right this should just trigger for subnet types not for node selections ?
> and in an subnet the inputs are determined by the input nodes that have a connection and the outputs by the output nodes

Ah you may be right. :)

Reply:
> Could you add a comment above this as to what it is trying to solve?
> 
> Why do we care about the children output nodes? Also, doesn't `allSubChildren()` recurse into potential other subnetworks and their children, etc. which doesn't sound like "outputs" we care about?

yes should have been .children() not allSubChildren(). 
i fixed this in the last PR